tests/unit/test_space.py

```diff
@@ -423,3 +423,121 @@ def test_box_deepcopy() -> None:
     box_copy = copy.deepcopy(box)
     npt.assert_allclose(box.lower, box_copy.lower)
     npt.assert_allclose(box.upper, box_copy.upper)
+
+
+@pytest.mark.parametrize(
+    "lower, upper, stepsizes",
+    [
+        pytest.param([0.0, 1.0], [1.0, 2.0], [0.1, 0.2], id="lists"),
+        pytest.param((0.0, 1.0), (1.0, 2.0), (0.1, 0.2), id="tuples"),
+        pytest.param(range(2), range(1, 3), [0.1, 0.2], id="ranges"),
+    ],
+)
+def test_ordinalsearchspace_converts_sequences_to_float64_tensors(lower, upper, stepsizes) -> None:
+    ordinalsp = OrdinalSearchSpace(lower, upper, stepsizes)
+    assert tf.as_dtype(ordinalsp.lower.dtype) is tf.float64
+    assert tf.as_dtype(ordinalsp.upper.dtype) is tf.float64
+    assert tf.as_dtype(ordinalsp.stepsizes.dtype) is tf.float64
+    npt.assert_array_equal(ordinalsp.lower, [0.0, 1.0])
+    npt.assert_array_equal(ordinalsp.upper, [1.0, 2.0])
+    npt.assert_array_equal(ordinalsp.stepsizes, [0.1, 0.2])
+
+
+@pytest.mark.parametrize(
+    "lower, upper, stepsizes",
+    [
+        pytest.param([0.0, 1.0], [1.0, 2.0], [0.1, 0.2]),
+        pytest.param([0.0, 1.0, 2.0], [1.0, 2.0, 3.0], [0.1, 0.2, 0.3]),
+        pytest.param([1.0, 10.0], [10.0, 100.0], [2, 10]),
+    ],
+)
```

Comment:
Does this work for step size that is a fraction?

Reply:
not sure it will, using fraction is quite risky since there will be a chance that a floating point error present on the result, hence decimal is preferred.

Reply:
Sorry, I meant decimal. Doesn't seem to be tested? so have a step size of 0.1 for example

Reply:
ah as you see there the couple first test is indeed using decimal

Reply:
1.0, 10.0 and 100.0 are still integers. I mean a step size of fraction (expressed as a decimal), like 0.1 or 0,2. i.e. we might want a space of [0,0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1]

Reply:
yes it's on line 449 and 450. the stepsizes are the third list of the pytest.param
```
        pytest.param([0.0, 1.0], [1.0, 2.0], [0.1, 0.2]),
        pytest.param([0.0, 1.0, 2.0], [1.0, 2.0, 3.0], [0.1, 0.2, 0.3]),
```
is this what you mean?

Reply:
OFC. Sorry was being stupid

Reply:
No problem